qualtran/bloqs/block_encoding/chebyshev_polynomial.py

```diff
@@ -75,6 +76,10 @@ class ChebyshevPolynomial(BlockEncoding):
     def __attrs_post_init__(self):
         if self.order < 0:
             raise ValueError(f"order must be greater >= 0. Found {self.order}.")
+        if not isinstance(self.block_encoding.signal_state.prepare, PrepareIdentity):
+            raise ValueError(
+                "Cannot take Chebyshev polynomial of block encodings with non-zero signal state."
+            )
```

Comment:
Is this also the case for `ScaledChebyshevPolynomial`?

Reply:
It is captured when `ScaledChebyshevPolynomial` instantiates `ChebyshevPolynomial` with the same arguments.

I added the redundant check here though for clarity.